prov/gni/src/gnix_dom.c

```diff
@@ -95,6 +96,9 @@ static int gnix_domain_close(fid_t fid)
 		/* TODO: free nic here */
 	}
 
+	assert(atomic_get(&domain->fabric->ref_cnt) > 0);
+	atomic_dec(&domain->fabric->ref_cnt);
+
```

Comment:
So if I understand you correctly, move it after and change the comparison to be >= 0?